ckanext/matomo/commands.py

```diff
@@ -149,3 +174,42 @@ def init_db():
     from ckanext.matomo.model import init_tables
     import ckan.model as model
     init_tables(model.meta.engine)
+
+
+def migrate():
+    """ Adds any missing columns to the database table for ResourceStats by
+        checking the schema and adding those that are missing.
+
+        If you wish to add a column, add the column name and sql
+        statement to MIGRATIONS_ADD which will check that the column is
+        not present before running the query.
+
+        If you wish to modify or delete a column, add the column name and
+        query to the MIGRATIONS_MODIFY which only runs if the column
+        does exist.
+        """
+    from ckan import model
+
+    MIGRATIONS_ADD = OrderedDict({
+        "downloads": "ALTER TABLE resource_stats ADD COLUMN downloads INT",
+    })
+
+    MIGRATIONS_MODIFY = OrderedDict({
+    })
+
+    q = "SELECT column_name from INFORMATION_SCHEMA.COLUMNS where table_name = 'resource_stats';"
+    current_cols = list([m[0] for m in model.Session.execute(q)])
+    for k, v in MIGRATIONS_ADD.items():
+        if k not in current_cols:
+            log.info(u"Adding column '{0}'".format(k))
+            log.info(u"Executing '{0}'".format(v))
+            model.Session.execute(v)
+            model.Session.commit()
+
+    for k, v in MIGRATIONS_MODIFY.items():
+        if k in current_cols:
+            log.info(u"Removing column '{0}'".format(k))
+            log.info(u"Executing '{0}'".format(v))
+            model.Session.execute(v)
+            model.Session.commit()
+    log.info("Migrations complete")
```

Comment:
This should be done using alembic, example here https://docs.ckan.org/en/2.9/extensions/best-practices.html?highlight=alembic#use-migrations-when-introducing-new-models

Reply:
Updated migrations to use alembic